ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBackupManager.java

```diff
@@ -36,14 +36,12 @@
 
   public CloudBackupManager(CloudConfig cloudConfig, ReplicationConfig replicationConfig,
       ClusterMapConfig clusterMapConfig, StoreConfig storeConfig, StoreKeyFactory storeKeyFactory,
-      ClusterMap clusterMap, VirtualReplicatorClusterMap virtualReplicatorClusterMap,
-      ScheduledExecutorService scheduler, ConnectionPool connectionPool, MetricRegistry metricRegistry,
-      NotificationSystem requestNotification, StoreKeyConverterFactory storeKeyConverterFactory,
-      String transformerClassName) throws ReplicationException {
+      ClusterMap clusterMap, VirtualReplicatorCluster virtualReplicatorCluster, ScheduledExecutorService scheduler,
+      ConnectionPool connectionPool, MetricRegistry metricRegistry, NotificationSystem requestNotification,
+      StoreKeyConverterFactory storeKeyConverterFactory, String transformerClassName) throws ReplicationException {
 
     super(replicationConfig, clusterMapConfig, storeKeyFactory, clusterMap, scheduler,
-        virtualReplicatorClusterMap.getDataNodeId(clusterMapConfig.clusterMapHostName,
-            clusterMapConfig.clusterMapPort), connectionPool, metricRegistry, requestNotification,
+        virtualReplicatorCluster.getDataNodeId(), connectionPool, metricRegistry, requestNotification,
         storeKeyConverterFactory, transformerClassName);
   }
```

Comment:
Feel free to add the rest of the constructor from your gist into this PR, unless it's not ready?

Reply:
Unfortunately, they are not ready :(. We need CloudDataNode, CloudReplica and a CloudDestinationFactory to produce CloudDestination.